solutions/usaco-1233.mdx

```diff
@@ -0,0 +1,76 @@
+---
+id: usaco-1233
+source: USACO Gold 2022 Open
+title: Apple Catching
+author: Qi Wang
+---
+
+[Official Analysis (Java)](http://www.usaco.org/current/data/sol_prob1_gold_open22.html)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N \log N)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+int N, ans = 0;
+map<int, int> mpp;
+struct item {
```

Comment:
rename to `Apple`

Reply:
technically can be a cow too

Reply:
then rename to `Item` and give some better field names

Reply:
i just used the variable letters stated in the problem

Reply:
at least add a comment next to each